File: Decorator/PHP/DarkRoast.php
<?php
  class DarkRoast extends Beverage {
    public function __construct() {
      parent::__construct();
      $this->description = " DarkRoast";
    }

    public function cost() {
      return 5.0;
    }

    public function getDescription() {
      return $this->description;
    }
  }

File: Decorator/Go/decorator/darkroast.go
package decorator

type DarkRoast struct {
	*Coffee
}

func NewDarkRoast() *DarkRoast {
	return &DarkRoast{&Coffee{price: 2.0, description: "DarkRoast"}}
}

func (self *DarkRoast) Cost() float64 {
	return self.price
}

func (self *DarkRoast) GetDescription() string {
	return self.description
}
